packages/calcite-components/src/components/meter/meter.tsx

```diff
@@ -0,0 +1,543 @@
+
+  @State() highPercent: number;
+
+  @State() lowActive: boolean;
+
+  @State() lowPercent: number;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  //--------------------------------------------------------------------------
+
+  private resizeHandler(): void {
+    this.updateLabels();
+  }
+
+  private updateLabels(): void {
+    if (this.valueLabelEl) {
+      this.determineValueLabelPosition();
+    }
+    if (this.rangeLabels) {
+      this.determineVisibleLabels();
+    }
+  }
+
+  private calculateValues(): void {
+    const { min, max, low, high, value } = this;
+    const lowPercent = (100 * (low - min)) / (max - min);
+    const highPercent = (100 * (high - min)) / (max - min);
+    const currentPercent = (100 * (value - min)) / (max - min);
+
+    if (!low || low < min || low > high || low > max) {
+      this.low = min;
+    }
+    if (!high || high > max || high < low || high < min) {
+      this.high = max;
+    }
+    if (!value) {
+      this.value = min;
+    }
+
+    this.lowPercent = lowPercent;
+    this.highPercent = highPercent;
+    this.currentPercent = value ? currentPercent : 0;
+    this.lowActive = !!low && low > min && (!value || low > value) && (!high || low < high);
+    this.highActive =
+      !!high && min <= high && high < max && (!value || high > value) && (!low || high > low);
+  }
+
+  /**
+   *
+   * @param value
+   * @param labelType
+   * @returns
+   */
+  private formatLabel = (value: number, labelType: MeterLabelType): string => {
+    if (labelType === "percent") {
+      if (!this.percentFormatting) {
+        const locale = getSupportedLocale(this.effectiveLocale);
+        const formatter = new Intl.NumberFormat(locale, {
+          useGrouping: this.groupSeparator,
+          style: "percent",
+        });
+        this.percentFormatting = { formatter, locale };
+      }
+      return this.percentFormatting.formatter.format(value);
+    } else {
+      numberStringFormatter.numberFormatOptions = {
+        locale: this.effectiveLocale,
+        numberingSystem: this.numberingSystem,
+        useGrouping: this.groupSeparator,
+      };
+      return numberStringFormatter.localize(value.toString());
+    }
+  };
+
+  private getMeterKindCssClass(): string {
+    const { low, high, min, max, value } = this;
+    const lowest = low ? low : min;
+    const highest = high ? high : max;
+    const aboveLowest = value >= lowest;
+    const belowLowest = value < lowest;
+    const aboveHighest = value >= highest;
+    const belowHighest = value < highest;
+
+    if (!value || (!low && belowHighest) || belowLowest) {
+      return CSS.success;
+    } else if (aboveLowest && belowHighest) {
+      return CSS.warning;
+    } else if (aboveHighest) {
+      return CSS.danger;
+    } else {
+      return CSS.success;
+    }
+  }
+
+  private intersects(el1: HTMLDivElement, el2: HTMLDivElement): boolean {
+    return el1 && el2
```

Comment:
Suggestion: 

```ts
return el1 && el2 && intersects(el1.getBoundingClientRect(), el2.getBoundingClientRect());
```

You already guard against elements being defined, so no need for optional chaining. Sidebar: I think we could also improve the util to handle null to simplify this type of logic.  🤓